deploy/helm/metrics-collector-config.yaml

```diff
@@ -1170,5 +1170,10 @@ service:
 {{- end }}
 {{- if .Values.otel.metrics.telemetry.metrics.enabled }}
     metrics:
-      address: {{ .Values.otel.metrics.telemetry.metrics.address }}
+      readers:
+        - pull:
+            exporter:
+              prometheus:
+                host: {{ (split ":" .Values.otel.metrics.telemetry.metrics.address)._0 | quote }}
```

Comment:
I'm not fan of the way we pass these telemetry settings to the OTEL Collector. I'd rather pass them as-is. And use this parsing logic only when the customer used the deprecated `address` setting. We even say that the documentation for the available telemetry settings is the official one from the OTEL Collector.